core/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/persistence/manager/ReferablePersistenceManager.java

```diff
@@ -148,8 +146,15 @@ else if (SubmodelElementCollection.class.isAssignableFrom(referable.getClass()))
                     Submodel.class,
                     SubmodelElementCollection.class));
         }
-        submodelElements.removeIf(x -> x.getIdShort().equalsIgnoreCase(submodelElement.getIdShort()));
-        submodelElements.add(submodelElement);
+        Predicate<SubmodelElement> filter = x -> x.getIdShort().equalsIgnoreCase(submodelElement.getIdShort());
+        int idx = List.class.isAssignableFrom(submodelElements.getClass()) ? ((List<SubmodelElement>) submodelElements)
```

Comment:
refactor to new method in CollectionHelper class `public static <T> void replace(Collection<T> collection, T old, T new)`